### PR TITLE
describe pre-commit + pre-commit.ci

### LINKED ISSUE #75: Add notes about pre-commit
In `content/15.development_procedure.md` we need some words about pre-commit.ci.

### PR DESCRIPTION
**Pull request by neutrinoceros**
fix #75
I attempted to slightly rearrange the whole section to preserve a sense of logical flow.
In doing so, I through in a mention of mypy, but it's a question at this point since the corresponding PR hasn't been accepted yet, so I'm going to mark this as a draft for now.
@matthewturk, if you wish I can separated this as two PRs, the first of which would exclude the third commit and be ready to go.

### CODE REVIEW

File: content/15.development_procedure.md [hunk 1]
```diff
@@ -92,15 +92,26 @@ By queuing up pull requests for merging, it allows maintainers to mark a set of
...
 
-For the purposes of checking for common code mistakes, we apply [``flake8``](https://flake8.pycqa.org/en/latest/), which reports unused variables, undefined variables, and other style violations that cannot be automatically corrected.
+We rely on the [``pre-commit``](https://pre-commit.com) framework, which enables automated checks as well as automatic fixes at commit time.
+This tool is an opt-in so not every "drive-by" contributor has to learn and install it, but continuous linting is provided by [pre-commit.ci](https://pre-commit.ci) so styling errors cannot slip in.
 
-All changes that can be applied via automation (specifically, code formatting) are accessible from within the GitHub pull request interface.
-This allows "drive-by" contributions to have their pull requests updated inline by an automated process, reducing the need to manually install packages to apply the changes.
+We configure `pre-commit` to run a mixed bag of formatters and static checkers.
```
Comment (matthewturk) [inline content/15.development_procedure.md:99]:
```suggestion
We configure `pre-commit` to run a mixture of formatters and static checkers.
```

("mixed bag" is often used slightly differently, to communicate a value judgment)

File: content/15.development_procedure.md [hunk 2]
```diff
@@ -92,15 +92,26 @@ By queuing up pull requests for merging, it allows maintainers to mark a set of
...
+The former modify the code in place, while the latter only report errors and code smells (such as undefined variables, unused imports, bare `except` statements...) but let human developers fix them.
```
Comment (matthewturk) [inline content/15.development_procedure.md:100]:
```suggestion
The former modify the code in place, while the latter only report errors and so-called "code smells" (such as undefined variables, unused imports, bare `except` statements...) but require human developers fix them.
```

File: content/15.development_procedure.md [hunk 3]
```diff
@@ -92,15 +92,26 @@ By queuing up pull requests for merging, it allows maintainers to mark a set of
...
+Our suite of formatter most proeminently includes
+[``black``](https://black.readthedocs.io/en/stable/),
+[``isort``](https://pycqa.github.io/isort/) and
+[``pyupgrade``](https://github.com/asottile/pyupgrade).
+``black`` has been designed to maximize readability with as few free parameters as possible
+(In many ways, the fact that most of the ``yt`` code developers did not utilize this style before it was enforced likely enabled its uptake, as it was seen as a choice that "made everyone compromise."),
+while ``isort`` ensures that all ``import`` statements are sorted (according to alphabetical order within a first/second/third-party categorization), and ``pyupgrade`` modernizes some Python idioms according to our minimal support version of the language.
+In particular, ``pyupgrade`` enforces the usage of modern Python ["f-strings"](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) since we do not support Python version older than 3.6.
+
+For static code analysis we rely on the [``flake8``](https://flake8.pycqa.org/en/latest/) framework (an possibly mypy for the specialized task of type checking ??)
```
Comment (matthewturk) [inline content/15.development_procedure.md:111]:
```suggestion
For static code analysis we rely on the [``flake8``](https://flake8.pycqa.org/en/latest/) framework.
Effort is underway to enable using `mypy` for the specialized task of type checking, but this is far from completion.
```

File: content/15.development_procedure.md (hunk 1 above)
Comment (neutrinoceros) [reply content/15.development_procedure.md:99]: I thought it did, but wasn't sure, thanks for having my back !

File: content/15.development_procedure.md (hunk 3 above)
Comment (neutrinoceros) [reply content/15.development_procedure.md:111]: actually not that far. Merging https://github.com/yt-project/yt/pull/3546 would suffice making yt "type-checked". Of course there's a lot more that we can do from there to _improve typing_, but as long as we are able to run mypy without error, we're "type-checked" !

Comment (matthewturk) [reply content/15.development_procedure.md:111]: I agree

**Comment by matthewturk** [comment]
This looks good -- can we toss it into ready-for-review state?

**Comment by neutrinoceros** [comment]
@matthewturk done !

**Pull request merged**